### PR TITLE
Feat/custom ldap mail attribute

### PR DESCRIPTION
**Pull request by dorianim**
Hi,

I created this based on the discussion [over here](https://github.com/directus/directus/discussions/9294).
Does not really do much, just add the config option `AUTH_<PROVIDER>_MAIL_ATTRIBUTE` which allows users to import mail addresses from a field other than `mail`.
What do you think? :slightly_smiling_face: 

Regards,
Dorian

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
Signaling LDAP-maestro @aidenfoxx 🙇🏻‍♂️

File: api/src/auth/drivers/ldap.ts [hunk 1]
```diff
@@ -136,7 +137,10 @@ export class LDAPAuthDriver extends AuthDriver {
 						const user = {
 							firstName: typeof object.givenName === 'object' ? object.givenName[0] : object.givenName,
 							lastName: typeof object.sn === 'object' ? object.sn[0] : object.sn,
-							email: typeof object.mail === 'object' ? object.mail[0] : object.mail,
+							email:
+								typeof object[mailAttribute ?? 'mail'] === 'object'
+									? object[mailAttribute ?? 'mail'][0]
+									: String(object[mailAttribute ?? 'mail']),
```
Comment (aidenfoxx) [inline api/src/auth/drivers/ldap.ts:143]: Pls break `object[mailAttribute ?? 'mail']` out into a variable. Also, don't cast to string. Undefined is a valid response for this parameter.

Comment (dorianim) [reply api/src/auth/drivers/ldap.ts]: Alright :upside_down_face:

**Comment by aidenfoxx** [comment]
@rijkvanzanten Looks good to me!

**Review by rijkvanzanten — APPROVED** [review]

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**